src/main/java/uk/gov/pay/api/agreement/service/AgreementsService.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.pay.api.agreement.service;
+
+import org.apache.http.HttpStatus;
+import uk.gov.pay.api.agreement.model.Agreement;
+import uk.gov.pay.api.agreement.model.AgreementCreatedResponse;
+import uk.gov.pay.api.agreement.model.AgreementLedgerResponse;
+import uk.gov.pay.api.agreement.model.AgreementSearchResults;
+import uk.gov.pay.api.agreement.model.CreateAgreementRequest;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.exception.CancelAgreementException;
+import uk.gov.pay.api.exception.CreateAgreementException;
+import uk.gov.pay.api.ledger.model.AgreementSearchParams;
+import uk.gov.pay.api.ledger.model.SearchResults;
+import uk.gov.pay.api.model.search.PaginationDecorator;
+import uk.gov.pay.api.service.ConnectorService;
+import uk.gov.pay.api.service.LedgerService;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+import java.util.stream.Collectors;
+
+public class AgreementsService {
+    private static final String AGREEMENTS_PATH = "/v1/agreements";
+    private final ConnectorService connectorService;
+    private final LedgerService ledgerService;
+    private final PaginationDecorator paginationDecorator;
+
+    @Inject
+    public AgreementsService(ConnectorService connectorService, LedgerService ledgerService, PaginationDecorator paginationDecorator) {
+        this.connectorService = connectorService;
+        this.ledgerService = ledgerService;
+        this.paginationDecorator = paginationDecorator;
+    }
+
+    public AgreementCreatedResponse createAgreement(Account account, CreateAgreementRequest createAgreementRequest) {
+        var response = connectorService.createAgreement(account, createAgreementRequest);
+
+        if (response.getStatus() != HttpStatus.SC_CREATED) {
+            throw new CreateAgreementException(response);
+        }
+        return response.readEntity(AgreementCreatedResponse.class);
+    }
+
+    public Response cancelAgreement(Account account, String agreementId) {
+        Response connectorResponse = connectorService.cancelAgreement(account, agreementId);
+        if (connectorResponse.getStatus() != HttpStatus.SC_NO_CONTENT) {
+            throw new CancelAgreementException(connectorResponse);
+        }
+
+        connectorResponse.close();
+        return Response.noContent().build();
```

Comment:
Not for this PR. This method can return void instead of Response. I think it is ideal if the resource method does `Response.noContent().build();`.

Reply:
👍